packages/gator-permissions-snap/src/core/permissionHandler.ts

```diff
@@ -206,6 +291,36 @@ export class PermissionHandler<
         await rerender();
       };
 
+      const accountSelectedHandler: UserEventHandler<
+        UserInputEventType.InputChangeEvent
+      > = async ({ event: { value } }) => {
+        const {
+          addresses: [address],
+        } = value as unknown as {
+          addresses: [`${string}:${string}:${string}`];
+        };
+
+        currentContext = {
+          ...currentContext,
+          accountAddressCaip10: address,
+        };
+
+        this.#tokenBalance = undefined;
+        this.#tokenBalanceFiat = undefined;
+
+        // we explicitly don't await this as it's a background process that will re-render the UI once it is complete
+        fetchAccountBalance(currentContext).catch(console.error);
```

Comment:
I think maybe we should use `logger.error` instead of `console.error` here for consistency and better log management.

 ```ts
 fetchAccountBalance(currentContext).catch((error) => {
     logger.error('Failed to fetch account balance', error);
     // Consider showing the error status to the user.
 });
 ```

Reply:
👍 addressed and commented above!